### DISCUSSION TITLE
display video inside a frame in a UI

### DISCUSSION THREAD

**Discussion by maddoxx099** [TomSchimansky/CustomTkinter#1116] [Q&A]
I have been trying to write down a function which takes video input from user and previews the video inside a preview frame but have not been able to figure out how to do. can anyone help me out on it.
For a  static image i had used this method and it seems to not work with video inputs

i am able to display the video in a seperate window but am  not able to display it in a frame in an already made UI

current code -

    def video_button_event(self):
        print("File works")
        filename = filedialog.askopenfilename(title="Select Video file")
        abs_path = os.path.abspath(filename)
        self.preview_video(abs_path)

    def preview_video(self, path):
        cap = cv2.VideoCapture(path)
        while (cap.isOpened()):
            ret, frame = cap.read()
            if ret:
                img = cv2.resize(frame, (400, 350))
                button_image = customtkinter.CTkImage(img, size=(400, 350))
                customtkinter.CTkButton(self.display_frame, text="", image=button_image, width=400, height=350).grid(
                    row=1, column=0, rowspan=2, padx=0, pady=0)
                if cv2.waitKey(25) & 0xFF == ord('q'):
                    break
            else:
                break
        cap.release()
        cv2.destroyAllWindows()

full code also attached if anyone can figure it out from that
[UI.txt](https://github.com/TomSchimansky/CustomTkinter/files/10499937/UI.txt)

**Comment by Akascape** [comment]
@maddoxx099 You can use tkvideoplayer to show video files.
Here is an example: https://github.com/TomSchimansky/CustomTkinter/discussions/531#discussioncomment-3922292
